src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_pending_resumption_handler.cc

```diff
@@ -0,0 +1,209 @@
+
+    SDL_LOG_DEBUG("Subscribing for event with function id: "
+                  << fid << " correlation id: " << cid);
+    subscribe_on_event(fid, cid);
+    pending_requests_[cid] = *request;
+    SDL_LOG_DEBUG("Sending request with fid: " << fid << " and cid: " << cid);
+    application_manager_.GetRPCService().ManageHMICommand(request);
+  }
+}
+
+void SDLPendingResumptionHandler::RaiseFakeSuccessfulResponse(
+    ns_smart_device_link::ns_smart_objects::SmartObject response,
+    const int32_t corr_id) {
+  using namespace application_manager;
+  response[strings::params][strings::correlation_id] = corr_id;
+  auto fid = static_cast<hmi_apis::FunctionID::eType>(
+      response[strings::params][strings::function_id].asInt());
+  event_engine::Event event(fid);
+  event.set_smart_object(response);
+
+  SDL_LOG_TRACE("Raise fake response for subscriber. corr_id : " << corr_id);
+  event.raise(application_manager_.event_dispatcher());
+}
+
+void SDLPendingResumptionHandler::on_event(
+    const application_manager::event_engine::Event& event) {
+  using namespace application_manager;
+  SDL_LOG_AUTO_TRACE();
+
+  const smart_objects::SmartObject& response = event.smart_object();
+  const uint32_t corr_id = event.smart_object_correlation_id();
+
+  SDL_LOG_TRACE("Received event with function id: "
+                << event.id() << " and correlation id: " << corr_id);
+
+  smart_objects::SmartObject pending_request;
+  if (pending_requests_.find(corr_id) == pending_requests_.end()) {
+    SDL_LOG_ERROR("corr id " << corr_id << " NOT found");
+    return;
+  }
+  pending_request = pending_requests_[corr_id];
+  pending_requests_.erase(corr_id);
+  if (app_ids_.empty()) {
+    SDL_LOG_ERROR("app_ids is empty");
+    return;
+  }
+  uint32_t app_id = app_ids_.front();
+  app_ids_.pop();
+  auto app = application_manager_.application(app_id);
+  if (!app) {
+    SDL_LOG_ERROR("Application not found " << app_id);
+    return;
+  }
+
+  if (resumption::IsResponseSuccessful(response)) {
+    SDL_LOG_DEBUG("Resumption of subscriptions is successful");
+
+    application_manager_.SubscribeAppForWayPoints(app);
+
+    for (auto& freezed_resumption : frozen_resumptions_) {
+      auto corr_id = freezed_resumption.request_to_send_
+                         .message[strings::params][strings::correlation_id]
+                         .asInt();
+      RaiseFakeSuccessfulResponse(response, corr_id);
+      application_manager_.SubscribeAppForWayPoints(freezed_resumption.app_id);
+    }
+    frozen_resumptions_.clear();
+  } else {
+    SDL_LOG_DEBUG("Resumption of subscriptions is NOT successful");
+
+    if (frozen_resumptions_.empty()) {
+      SDL_LOG_DEBUG("freezed resumptions is empty");
+      return;
+    }
+
+    ResumptionAwaitingHandling freezed_resumption = frozen_resumptions_.back();
+    frozen_resumptions_.pop_back();
+    auto resumption_req = freezed_resumption.request_to_send_;
```

Comment:
```suggestion
    auto resumption_req = frozen_resumption.request_to_send_;
```

Reply:
@iCollin renamed in https://github.com/smartdevicelink/sdl_core/pull/3475/commits/8913095240e15b52211f29fb6af06a4bd573155a